test_pytest_rerunfailures.py

```diff
@@ -361,6 +362,125 @@ def test_fail_two():
     assert_outcomes(result, passed=0, failed=1, rerun=2)
 
 
+@pytest.mark.parametrize(("factor", "exp_base", "exp_max", "expected"), [
+    (-1, 2, None, [0, 0, 0]),     # Negative factor, apply default 0.
+    (3, 2, None, [3, 6, 12]),
+    (3, None, None, [3, 6, 12]),  # Missing base, apply default 2.
+    (3, 2, 7, [3, 6, 7]),
+    (3, 2, -1, [3, 6, 12]),       # Negative max, not considered.
+    (3, -1, None, [3, 6, 12]),    # Negative base, apply default 2.
+    (3, 3, None, [3, 9, 27]),
+])
+def test_reruns_backoff(testdir, monkeypatch, factor, exp_base,
+                        exp_max, expected):
+    testdir.makepyfile(
+        """
+        def test_fail():
+            assert False"""
```

Comment:
nit: if you allow a trailing `\n` in the file, it would look nicer to have the `"""` on the next line

Reply:
Done